src/storage/ArrayStorage.java

```diff
@@ -6,72 +6,103 @@
 import java.util.Objects;
 import java.util.logging.Logger;
 
-public class ArrayStorage {
-    private static final Logger log = Logger.getLogger("ArrayStorage");
-    private static final int NOT_EXISTING_INDEX = -1;
+import static java.util.logging.Level.WARNING;
 
-    private static final int MAX_STORAGE_SIZE = 100000;
-    private final Resume[] storage = new Resume[MAX_STORAGE_SIZE];
-    private int size;
+public abstract class ArrayStorage implements Storage {
+    protected static final int NOT_EXISTING_INDEX = -1;
+    protected static final int MAX_STORAGE_SIZE = 100000;
 
+    protected final Logger log = Logger.getLogger(getClass().getSimpleName());
+
+    protected Resume[] storage;
+
+
+    protected ArrayStorage() {
+        this.storage = new Resume[MAX_STORAGE_SIZE];
+    }
+
+    protected int size;
+
+    @Override
     public void clear() {
         Arrays.fill(storage, 0, size, null);
         size = 0;
     }
 
+    @Override
     public void save(Resume r) {
         Objects.requireNonNull(r);
         if (size == MAX_STORAGE_SIZE) {
-            log.warning("Storage max limit is reached. Please, clear storage to add new elements.");
+            logStorageMaxLimit();
             return;
         }
         if (isPresent(r.getUuid())) {
```

Comment:
There is no need for `getResumeIndex()` method invocation twice. It's called in the `isPresent()` and in the code below

Reply:
Done